documentation/troubleshooting.md

```diff
@@ -0,0 +1,90 @@
+```
+
+if you are unable to find controller-manager pod, use `docker ps -a` as below to grab `Container ID`.
+
+```
+# docker ps | grep controller-manager
+74a15e75d136        gcr.io/google_containers/kube-controller-manager-amd64   "kube-controller-m..."   23 minutes ago      Up 23 minutes                           k8s_kube-controller-manager_kube-controller-manager-kubeadm-master_kube-system_22d908089b353bf8749a89843022bff3_0
+12df71e6d4a8        gcr.io/google_containers/pause-amd64:3.0                 "/pause"                 23 minutes ago      Up 23 minutes                           k8s_POD_kube-controller-manager-kubeadm-master_kube-system_22d908089b353bf8749a89843022bff3_0
+
+
+
+# docker inspect 74a15e75d136 | grep Id
+        "Id": "74a15e75d1365164cec2c005030a7957ddb3d3c9ee487c6b40df35b0a4e4f95a",
+```
+Once the `Container ID` is obtained, logs can be obtained from `/var/lib/docker/containers/` directory. Docker creates the directory for each container and keeps logs in that directory.
+
+```
+# ls /var/lib/docker/containers/74a15e75d1365164cec2c005030a7957ddb3d3c9ee487c6b40df35b0a4e4f95a
+74a15e75d1365164cec2c005030a7957ddb3d3c9ee487c6b40df35b0a4e4f95a-json.log  checkpoints  config.v2.json  hostconfig.json
+```
+
+**Note:** On `systemd-based` setup, you may need to use `journalctl` instead. You need to know the service name to get the logs for controller-manager.
+- Here is the example.
+  ```
+  journalctl -u kube-controller-manager.service > controller-manager.log
+  ```
+
+Kubelet generally runs as the service and not in the container or Pod in the Kubernetes cluster. Kubelet logs can be obtained using `journalctl` as shown below.
+
+```
+journalctl -u kubelet > kubelet.log
+```
+
+## How can I modify the log level?
+
+Log levels can be adjusted using `--v` option in the pod manifests files. To help debug vSphere Cloud Provider, we recommended increasing log level for Controller-Manager. Manifest files are generally located at `/etc/kubernetes/manifests/`.
+
+```
+# cd /etc/kubernetes/manifests/
+# ls
+etcd.yaml  kube-apiserver.yaml  kube-controller-manager.yaml  kube-scheduler.yaml
+```
+
+Open YAML/JSON file and find `--v` option in the container's command. Increase it to 9. if this option is not present then add --v=9 to the container's command for the controller-manager, so that we can help debug issues in the vSphere Cloud Provider.
+
+```
+# cat kube-controller-manager.yaml
+apiVersion: v1
+kind: Pod
+metadata:
+.
+.
+spec:
+  containers:
+  - command:
+    - kube-controller-manager
+    - --service-account-private-key-file=/etc/kubernetes/pki/sa.key
+    .
+    .
+    .
+    - --kubeconfig=/etc/kubernetes/controller-manager.conf
+    - --v=9
+```
+After increasing the log level, you need to restart the `kubelet`, which will restart the API server and Controller-Manager with updated pod manifest file.
+
+
+## What information is needed to provide to technical support for analysis?
```

Comment:
I am not sure why do we need this section given that logs are collected (mostly) for troubleshooting technical issues.

Reply:
Removed this section.